src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -851,7 +859,10 @@ public bool TakeMessageHeaders(SocketInput input, FrameRequestHeaders requestHea
                 while (!scan.IsEnd)
                 {
                     var beginName = scan;
-                    scan.Seek(ref _vectorColons, ref _vectorCRs);
+                    if (scan.Seek(ref _vectorColons, ref _vectorCRs) == -1)
+                    {
+                        return false;
```

Comment:
@halter73 Now I see that I read it wrong. But as you point out it's still an issue not checking it's return value :grin: